### PR TITLE
caffe: fixed build error

### PR DESCRIPTION
**Pull request by tkameyama**
- caffe depemndents hdf5 +hl +cxx
- change caffe dependency of opencv
- opencv 3.2 required old cmake

### CODE REVIEW

File: var/spack/repos/builtin/packages/caffe/package.py [hunk 1]
```diff
@@ -40,10 +40,10 @@ class Caffe(CMakePackage):
     depends_on('protobuf')
     depends_on('glog')
     depends_on('gflags')
-    depends_on('hdf5')
+    depends_on('hdf5 +hl +cxx')
 
     # Optional dependencies
-    depends_on('opencv@3.2.0+core+highgui+imgproc', when='+opencv')
+    depends_on('opencv@3.2.0+core+highgui+imgproc~vtk~videoio~gtk~python', when='+opencv')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/caffe/package.py]: `caffe` doesn't work if you link to `opencv+vtk...` or you just want to change the default?

File: var/spack/repos/builtin/packages/opencv/package.py [hunk 2]
```diff
@@ -112,6 +112,7 @@ class Opencv(CMakePackage, CudaPackage):
                  tag="{0}".format(cv),
                  when='@{0}+cuda'.format(cv))
 
+    depends_on('cmake@:3.11', when='@:3.2', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/opencv/package.py]: Can you explain this?

File: var/spack/repos/builtin/packages/caffe/package.py (hunk 1 above)
Comment (tkameyama) [reply var/spack/repos/builtin/packages/caffe/package.py]: openvc@3.2.0+vtk... did not build.
I'm backport to build openvc.

File: var/spack/repos/builtin/packages/opencv/package.py (hunk 2 above)
Comment (tkameyama) [reply var/spack/repos/builtin/packages/opencv/package.py]: cmake 3.12 is changed python detection code.
I'm backport for cmake 3.12 or later.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**